src/CodeGen_C.cpp

```diff
@@ -353,6 +395,777 @@ string type_to_c_type(Type type, bool include_space, bool c_plus_plus = true) {
         oss << " ";
     return oss.str();
 }
+
+}
+
+void CodeGen_C::add_vector_typedefs(const std::set<Type> &vector_types) {
+    if (!vector_types.empty()) {
+        // MSVC has a limit of ~16k for string literals, so split
+        // up these declarations accordingly
+        const char *cpp_vector_decl = R"INLINE_CODE(
+#if !defined(__has_attribute)
+    #define __has_attribute(x) 0
+#endif 
+
+#if !defined(__has_builtin)
+    #define __has_builtin(x) 0
+#endif 
+
+template <typename ElementType_, size_t Lanes_>
+class CppVector {
+public:
+    typedef ElementType_ ElementType;
+    static const size_t Lanes = Lanes_;
```

Comment:
Exposing the arguments to the template as internal-but-public typedefs allows other pieces of code to inspect the particular vector variant in question; the convert_from() implementation uses this to sanity-check the "OtherVec" type. (There were formerly a few other uses too, in code that didn't make the final cut.)